build.sh

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+set -ex
+if [[ "$TRAVIS_PULL_REQUEST" != "false" ]]; then     # on pull requests
+    mvn clean verify
+elif [[ -z "$TRAVIS_TAG" && "$TRAVIS_BRANCH" == "master" ]]; then  # non-tag commits to master branch
+    mvn clean deploy
+elif [[ -z "$TRAVIS_TAG" && "$TRAVIS_BRANCH" =~ ^stable-.* ]]; then # non-tag commits to stable branches
+    mvn clean deploy
```

Comment:
What is the goal here? If the goal is for Travis to run tests for each build, I believe mvn deploy includes verify. If it does not, we could simply change the script to mvn clean verify deploy.

Running tests on bridge-base is already a requirement for check-in, and thus far we haven't had any problems with it. There is no need to enforce it. If the build breaks, GitHub makes it easy to revert the pull request.

Reply:
>What is the goal here? If the goal is for Travis to run tests for each build, I believe mvn >deploy includes verify. If it does not, we could simply change the script to mvn clean >verify deploy.

The goal with this script is to have travis do different things on pushes/merges.
1. only run build & test on a PR
2. run bulid, test & deploy on merge to master branch.  This deploys SNAPSHOT version, i.e. "bridge-base-1.0.0-SNAPSHOT"
3. run build, test & deploy on change to stable branch.  This deploys a non-snapshot version "bridge-base-1.0.0"

>Running tests on bridge-base is already a requirement for check-in, and thus far we >haven't had any problems with it. There is no need to enforce it. If the build breaks, >GitHub makes it easy to revert the pull request.

This is a preference.  Doing pre-merge testing is a preventive measure.  Yes, people can run test on their own before merging but people are not real reliable.  I prefer to use the github PR build feature so that travis will test for us and travis will validates in the "official" build environment so we don't have to worry about things like missing committed files and whatnot.   So yes, it's pretty easy to revert if something bad was committed, pre-merge testing just helps to run test before something bad goes in.

Reply:
What is the problem we're trying to solve here? I'm not convinced this is worth the extra churn of extra Travis builds or the extra complexity of having a custom build script.